presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -69,22 +77,50 @@ public HivePageSourceProvider(
     @Override
     public ConnectorPageSource createPageSource(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorSplit split, List<ColumnHandle> columns)
     {
-        HiveSplit hiveSplit = checkType(split, HiveSplit.class, "split");
-
-        String clientId = hiveSplit.getClientId();
+        List<HiveColumnHandle> hiveColumns = ImmutableList.copyOf(transform(columns, HiveColumnHandle::toHiveColumnHandle));
```

Comment:
We don't use `transform` any more. Rewrite using streaming API. Don't worry about Immutable.